tests/test_authtoken.py

```diff
@@ -49,6 +51,63 @@ def test_whitespace_in_password(self):
         assert AuthTokenSerializer(data=data).is_valid()
 
 
+
+    def test_token_creation_collision_raises_integrity_error(self):
+        """
+        Verify that creating a token with an existing key raises IntegrityError.
+        """
+        user2 = User.objects.create_user('user2', 'user2@example.com', 'p')
+        existing_token = Token.objects.create(user=user2)
+
+        # Try to create another token with the same key
+        with self.assertRaises(IntegrityError):
+            Token.objects.create(key=existing_token.key, user=self.user)
+
+    def test_key_regeneration_on_save_is_not_a_breaking_change(self):
+        """
+        Verify that when a token is created without a key, it generates one correctly.
+        This tests the backward compatibility scenario where existing code might
+        create tokens without explicitly setting a key.
+        """
+        # Create a new user for this test to avoid conflicts with setUp token
+        user2 = User.objects.create_user('test_user2', 'test2@example.com', 'password')
+        
+        # Create a token without a key - it should generate one automatically
+        token = Token(user=user2)
+        token.key = ""  # Explicitly clear the key
+        token.save()
```

Comment:
This token is saved only once to the DB here. 

I think this test should:

1. Create a token, generating the associated key & saving to the DB
2. Clear the `key` field, save again
3. Refresh from DB 
4. Check that it's been regenerated (different `key` from the original), but same PK

Basically we want to test that the second saving doesn't insert a new row in the table